src/main/java/me/zilzu/mycoupon/api/controller/ControllerAdvisor.java

```diff
@@ -0,0 +1,34 @@
+package me.zilzu.mycoupon.api.controller;
+
+import me.zilzu.mycoupon.application.service.CouponValidException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(RuntimeException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public ExceptionResponse exceptionCheck(RuntimeException runtimeException) {
+        System.out.println(runtimeException.getMessage());
+        return new ExceptionResponse(HttpStatus.INTERNAL_SERVER_ERROR.toString(), runtimeException.getMessage());
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ExceptionResponse exceptionCheck(IllegalArgumentException illegalArgumentException) {
+        System.out.println(illegalArgumentException.getMessage());
+        return new ExceptionResponse(HttpStatus.BAD_REQUEST.toString(), illegalArgumentException.getMessage());
+    }
+
+    @ExceptionHandler(CouponValidException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ExceptionResponse exceptionCheck() {
```

Comment:
만약 CouponValidException 안에 데이터가 있으면, 여기서 

```suggestion
    public ExceptionResponse exceptionCheck(CouponValidException e) {
```

이런식으로 exception 받아서 안에 있는 데이터를 응답으로 내보는것도 가능!